Automattic/wp-calypso
# SMP: Hide stats the never finish loading

* Closes: https://github.com/Automattic/wp-calypso/issues/75072

## Proposed Changes

* This PR replaces the infinitely loading bar with the "Stats Off"  on the SMP page for sites that don't have stats enabled (some staging, Woo Express, and self-hosted sites on which the request to retrieve stats fails). 

## Testing Instructions

* Create a site with a `Woo Express: Trial (Monthly)` plan (you can add it to a site from Store Admin)
* Once the site is created, navigate to `https://wordpress.com/sites`. Make sure to select the table view
* Find the your `Woo Express: Trial` site and observe the **Stats** column
* Check that the infinitely loading stats bar is replaced with "Stats Off" text

<img width="1336" alt="Screenshot 2023-05-16 at 4 06 53 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/b364256f-2a64-4683-ba68-46dda7cf2c89">

* Hover over "Stats Off"
* Observe the popover appearing that says "Stats are disabled for this site"

<img width="1296" alt="Screenshot 2023-05-16 at 4 07 46 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/c1b9705d-db9c-401e-aeb9-8346a0f9bd1c">

* Hover away from "Stats Off"
* Observe the popover going away
* This view is disabled on mobile but is visible on some tablets. You can test with Chrome dev tools with iPad Air:

<img width="1617" alt="Screenshot 2023-05-16 at 4 13 11 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/6c90fe21-2306-48a0-b752-6a43178fcebe">

* Switch your account language from `me/account` to RTL (like Arabic) and check that the `Stats Off` indicator works as expected:

<img width="1607" alt="Screenshot 2023-05-16 at 4 17 01 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/3ba2d97d-be42-4e09-83bf-4c464c52033a">

<img width="1509" alt="Screenshot 2023-05-16 at 4 18 50 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/c9e8f052-f136-4683-b7e8-06aa66daa81c">

### Some Notes

* While working on this PR, I discovered that it is possible to enable the stats for Woo Express sites from this page `https://wordpress.com/stats/day/{yourecommercetrialsite}` and they appear under **Jetpack Stats**:

<img width="1543" alt="Screenshot 2023-05-16 at 3 23 16 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/4be4bffb-0f87-4bb5-8227-5957bbbb3703">

Once you activate them once from that page, they will be loaded on the SMP page (`https://wordpress.com/sites`) similar to all other sites and the loading bar will go away. I am not sure if it is intentional to allow stats on the trial site. If it is, should we link from the popover text "Stats are disabled for this site" to `https://wordpress.com/stats/day/` where the stats can be enabled?

## Pre-merge Checklist
